### ISSUE TITLE
Ora2pg error: FATAL: bad export type using input file option

### ISSUE THREAD

**Issue by naveenjul29** [darold/ora2pg#1057]
Hi Darold,

I am trying to export Materialized views using input file option and got the below error:

![Mview Capture](https://user-images.githubusercontent.com/67134199/103339190-be242b00-4aa6-11eb-9322-ce265fac6976.PNG)

Below query I used:

CREATE MATERIALIZED VIEW "MV_EMP" ("EMPNO", "ENAME", "JOB", "MGR", "HIREDATE", "SAL", "COMM", "DEPTNO") AS SELECT "EMP1"."EMPNO","EMP1"."ENAME","EMP1"."JOB","EMP1"."MGR","EMP1"."HIREDATE","EMP1"."SAL","EMP1"."COMM","EMP1"."DEPTNO" FROM "EMP1";

**Note**: same issue I am having with Synonyms also. 

Can you please let me know what are all the objects we cannot use with input file option?

**Comment by darold** [comment]
Yes this feature have not been implemented yet. Here are the export type that can be used from a file:
* TABLE
* SEQUENCE
* GRANT
* TABLESPACE
* VIEW
* TRIGGER
* QUERY
* FUNCTION
* PROCEDURE
* PACKAGE
* TYPE
* SYNONYM
* DIRECTORY
* DBLINK
* LOAD
I mark the issue as needing enhancement.

**Comment by naveenjul29** [comment]
Thank you Darold

**Comment by jkalta** [comment]
Hi Darold,
Any update on this enhancement?

**Comment by darold** [comment]
Hi, I have not planned to work on this on the near future.

**Comment by srinilakshmi** [comment]
@darold I am getting the same error for SHOW_REPORT too.

for SQL Server (scripts).

ora2pg -t SHOW_REPORT --dump_as_json -i "<input file>"

How do I get around with this?

**Comment by darold** [comment]
The -i option can not be used with the SHOW_REPORT action, SHOW_REPORT need a connection to an online database.

**Comment by srinilakshmi** [comment]
Thank you very much @darold. I have opened another issue [1828](https://github.com/darold/ora2pg/issues/1828) please let me know if you can. TIA
